tests/testsuite/install.rs

```diff
@@ -2730,3 +2730,20 @@ fn uninstall_running_binary() {
 
 "#]]).run();
 }
+
+#[cargo_test]
+fn dry_run() {
+    pkg("foo", "0.0.1");
```

Comment:
A case I think would be important for us to test for is `cargo install --dry-run some-package-from-the-distant-future` (we tend to use a rust_version of `"1.2345"`).

Ideally, we'd get the error about it being incompatible.

Its produced in `create_bcx` which looks like its called in dry-run mode, so it should be a simple change of adding the test to the test commit and updating it in the feature commit.  If its anything more, I'm fine with i being deferred out and tracked in #11123 as part of stabilization

Reply:
I rebased my two first commit to add this test in it